the-state-of-doctests-in-haskell/article.md

```diff
@@ -0,0 +1,253 @@
+# The state of doctests in Haskell
+
+### What are doctests?
+
+To put it simply, doctests are pieces of text within [haddock comments](https://www.haskell.org/haddock/) that look like interactive Haskell sessions.
+The idea to search for and execute those sessions all over Haskell modules actually comes from the awesome
+[`doctest`](https://en.wikipedia.org/wiki/Doctest) module, which is included in the [`Python programming language`](https://en.wikipedia.org/wiki/Python_(programming_language)).
+
+Documenting software is extremely challenging.
+The doctests concept makes the documentation process not only pleasant and effective - it actually allows Haskell functions to speak for themselves - but
+also helps to get the most out of your testing efforts.
+
+For the sake of simplicity and convenience, we won't cover the markup used to define a doctest.
```

Comment:
I think it would be more convenient if that would be covered. 

When I read this post, I want to know exactly how to define doctests and which library I want to use for this purpose.